pkg/retriever/bitswapretriever.go

```diff
@@ -53,12 +53,13 @@ type InProgressCids interface {
 // Note: this is a tradeoff over go-merkledag for traversal, cause selector execution is slow. But the solution
 // is to improve selector execution, not introduce unpredictable encoding.
 type BitswapRetriever struct {
-	bstore         MultiBlockstore
-	inProgressCids InProgressCids
-	routing        IndexerRouting
-	blockService   blockservice.BlockService
-	clock          clock.Clock
-	cfg            BitswapConfig
+	bstore                  MultiBlockstore
+	inProgressCids          InProgressCids
+	routing                 IndexerRouting
+	blockService            blockservice.BlockService
+	clock                   clock.Clock
+	cfg                     BitswapConfig
+	awaitReceivedCandidates chan<- struct{}
```

Comment:
this is leaky just for tests, right? or do we have another use for it? maybe a comment would be good to clarify its purpose

Reply:
just for tests.